pkg/controller/controller_test.go

```diff
@@ -0,0 +1,167 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/kubernetes-csi/external-resizer/pkg/csi"
+	"github.com/kubernetes-csi/external-resizer/pkg/resizer"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestController(t *testing.T) {
+	for _, test := range []struct {
+		Name string
+		PVC  *v1.PersistentVolumeClaim
+		PV   *v1.PersistentVolume
+
+		CreateObjects bool
+		NodeResize    bool
```

Comment:
should we add a field here for explicitly testing scenarios where we expect failure? As it is - we only test scenarios which aren't supposed to fail.

Reply:
All the failure scenarios I saw were either guarding against underlying kube problems (which I don't know how to mock) or resizer problems (which I believe should be handled in the resizer unit tests). Any advice on how of if either of those two categories should be tested would be appreciated.

Reply:
iirc - mock client supports adding reactors to certain actions on kube client. We could use that to test more custom stuff. For example, I was wondering if for test case where PV should be updated, if we can actually test if PV got updated. That would be awesome.